pkg/collectors/devices/dpll_netlink.go

```diff
@@ -100,42 +204,66 @@ func buildPostProcessDPLLNetlink(clockID *big.Int) fetcher.PostProcessFuncType {
 				processedResult[entry.ClockType] = state
 			}
 		}
+		pin := NetlinkPin{}
+		err = json.Unmarshal([]byte(result["dpll-netlink-offset"]), &pin)
+		if err != nil {
+			log.Errorf("Failed to unmarshal netlink pin output: %s", err.Error())
+		}
+
+		for _, parentPin := range pin.ParentDevices {
+			switch parentPin.ParentID % 2 {
+			case EECOffsetParentID:
+				processedResult["ecc_offset"] = parentPin.PhaseOffset
+			case PPSOffesetParentID:
+				processedResult["pps_offset"] = parentPin.PhaseOffset
+			}
+		}
 		return processedResult, nil
 	}
 }
 
-// BuildDPLLNetlinkInfoFetcher popluates the fetcher required for
+// BuildDPLLNetlinkDeviceFetcher popluates the fetcher required for
 // collecting the DPLLInfo
-func BuildDPLLNetlinkInfoFetcher(clockID *big.Int) error { //nolint:dupl // Further dedup risks be too abstract or fragile
+func BuildDPLLNetlinkDeviceFetcher(params NetlinkParameters) error { //nolint:dupl // Further dedup risks be too abstract or fragile
 	fetcherInst, err := fetcher.FetcherFactory(
 		[]*clients.Cmd{dateCmd},
 		[]fetcher.AddCommandArgs{
 			{
-				Key:     "dpll-netlink",
-				Command: "/linux/tools/net/ynl/cli.py --spec /linux/Documentation/netlink/specs/dpll.yaml --dump device-get",
-				Trim:    true,
+				Key: "dpll-netlink-device",
+				Command: "/linux/tools/net/ynl/cli.py --spec /linux/Documentation/netlink/specs/dpll.yaml --dump device-get | " +
```

Comment:
Instead of shelling out to cli.py, we could consider doing native netlink via https://pkg.go.dev/github.com/mdlayher/netlink (the same library that the linuxptp-daemon container uses).

Reply:
If we get to the point of generating the go structs though we could look to re-implement some of the python with specific commands that actually output proper json.